### PR TITLE
Cvs 139229 llm node resource

### PR DESCRIPTION
**Pull request by rasapala**
### 🛠 Summary

JIRA CVS-139229
Adding llm node resource class and initialization of <ContinuousBatchingPipeline> object.
Implementing llm node resources flow with graphdefinition and graphexecutor.
Adding node resource as inputSidePacket.
Initial implementation of llmcalculator to handle node resource.
Unit tests and DISABLED full flow functional test that requires llm models to run.
Fixing config.json removal.
Updating to required OV binary, model_api and openai SHAs.

### 🧪 Checklist

- [x] Unit tests added.
- [x] The documentation updated. - no documentation needed.
- [x] Change follows security best practices.
``

### CODE REVIEW

File: src/mediapipe_internal/mediapipegraphexecutor.hpp [hunk 1]
```diff
@@ -102,6 +104,8 @@ class MediapipeGraphExecutor {
 #if (PYTHON_DISABLE == 0)
         sideInputPackets[PYTHON_SESSION_SIDE_PACKET_TAG] = mediapipe::MakePacket<PythonNodeResourcesMap>(this->pythonNodeResourcesMap).At(STARTING_TIMESTAMP);
 #endif
+        if (this->llmNodeResourcesMap.size())
+            sideInputPackets[LLM_SESSION_SIDE_PACKET_TAG] = mediapipe::MakePacket<LLMNodeResourcesMap>(this->llmNodeResourcesMap).At(STARTING_TIMESTAMP);
```
Comment (mzegla) [inline src/mediapipe_internal/mediapipegraphexecutor.hpp:108]: We are you using different names here and in line 211?

File: src/test/llmnode_test.cpp [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+TEST_F(LLMFlowKfsTest, DISABLED_Infer) {
+    const ovms::Module* grpcModule = server.getModule(ovms::GRPC_SERVER_MODULE_NAME);
+    KFSInferenceServiceImpl& impl = dynamic_cast<const ovms::GRPCServerModule*>(grpcModule)->getKFSGrpcImpl();
+    ::KFSRequest request;
+    ::KFSResponse response;
+    const std::string modelName = "llmDummyKFS";
+    request.Clear();
+    response.Clear();
+    std::vector<std::string> requestData1{"What is OpenVINO?"};
+    std::string expectedResponse = "\n\nOpenVINO is an open-source software library for deep learning inference that is designed to optimize and run deep learning models on a variety";
```
Comment (mzegla) [inline src/test/llmnode_test.cpp:101]: Please add a comment for which model exactly this is the expected output

Comment (mzegla) [reply src/test/llmnode_test.cpp:101]: Model that was used: 
https://huggingface.co/meta-llama/Llama-2-7b-chat-hf
converted with optimum cli

File: src/llm/llmnoderesources.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+std::unordered_map<std::string, std::string> LLMNodeResources::prepareLLMNodeInitializeArguments(const ::mediapipe::CalculatorGraphConfig::Node& graphNodeConfig, std::string basePath) {
+    std::unordered_map<std::string, std::string> LLMArguments;
+    return LLMArguments;
+}
+
+Status LLMNodeResources::createLLMNodeResources(std::shared_ptr<LLMNodeResources>& nodeResources, const ::mediapipe::CalculatorGraphConfig::Node& graphNodeConfig, std::string graphPath) {
+    mediapipe::LLMCalculatorOptions nodeOptions;
+    graphNodeConfig.node_options(0).UnpackTo(&nodeOptions);
```
Comment (dkalinowski) [inline src/llm/llmnoderesources.cpp:50]: Are we sure that index 0 exists? Shouldnt we first check if node_options.size() > 0?

Comment (rasapala) [reply src/llm/llmnoderesources.cpp:50]: Its checked by the  caller with: PYTHON_NODE_MISSING_OPTIONS error.

File: src/test/llmnode_test.cpp [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+// --------------------------------------- OVMS LLM nodes tests
+// Test disabled by default - needs LLM models to work in /workspace directory:
+// openvino_detokenizer.bin  openvino_detokenizer.xml  openvino_model.bin  openvino_model.xml  openvino_tokenizer.bin  openvino_tokenizer.xml
+// Model that was used:
+// https://huggingface.co/meta-llama/Llama-2-7b-chat-hf
+// converted with optimum cli
+TEST_F(LLMFlowKfsTest, DISABLED_Infer) {
+    const ovms::Module* grpcModule = server.getModule(ovms::GRPC_SERVER_MODULE_NAME);
+    KFSInferenceServiceImpl& impl = dynamic_cast<const ovms::GRPCServerModule*>(grpcModule)->getKFSGrpcImpl();
+    ::KFSRequest request;
```
Comment (dkalinowski) [inline src/test/llmnode_test.cpp:98]: The LLM graphs will not be available via KServe API and will **not** accept KServe request. The graphs will be reachable only via OpenAI API. Please check how to set up HTTP OpenAI tests and mock http server streaming request: https://github.com/openvinotoolkit/model_server/pull/2436/files#diff-086971ad0a2af85fdbab17aeff70f49cbeacb9b943246aeb4add720a8312b506

File: src/test/llmnode_test.cpp [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+        }
+    )";
+
+    ovms::MediapipeGraphConfig mgc{"mediaDummy", "", ""};
+    DummyMediapipeGraphDefinition mediapipeDummy("mediaDummy", mgc, testPbtxt, nullptr);
+    mediapipeDummy.inputConfig = testPbtxt;
+    ASSERT_EQ(mediapipeDummy.validate(manager), StatusCode::LLM_NODE_DIRECTORY_DOES_NOT_EXIST);
+}
+
+TEST_F(LLMFlowKfsTest, LLMNodeBadWorkspacePathFile) {
```
Comment (dkalinowski) [inline src/test/llmnode_test.cpp:244]: What does it mean that it is bad?

File: src/test/llmnode_test.cpp [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+        }
+    )";
+
+    ovms::MediapipeGraphConfig mgc{"mediaDummy", "", ""};
+    DummyMediapipeGraphDefinition mediapipeDummy("mediaDummy", mgc, testPbtxt, nullptr);
+    mediapipeDummy.inputConfig = testPbtxt;
+    ASSERT_EQ(mediapipeDummy.validate(manager), StatusCode::LLM_NODE_DIRECTORY_DOES_NOT_EXIST);
+}
+
+TEST_F(LLMFlowKfsTest, LLMNodeResourceInitFailed) {
```
Comment (dkalinowski) [inline src/test/llmnode_test.cpp:269]: Why is it failed?

**Review by mzegla — APPROVED** [review]
Please also switch LLM build to this fork and branch:
https://github.com/mzegla/openvino.genai/tree/request_rate

File: src/test/llmnode_test.cpp (hunk 4 above)
Comment (mzegla) [reply src/test/llmnode_test.cpp:98]: I think for now, for the sake of internal testing we could leave it as is. gRPC could be a path for e2e testing until OpenAI is integrated.

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**